key requests require url-encoding. Also fix for SearchRequest

pupitetris opened the pull request:
url-encode the key parameter as it is going to be part of the request URL, which is particularly important for the Lit_ case (for example, when trying to send a Space).

All url-encoding calls, including that found in SearchRequest changed to use StandardCharsets.UTF_8.name() or otherwise a MethodNotFound exception was generated at runtime (maybe only on my development Android device, but this is more compatible in general). Also, UnsupportedEncodingException has to be handled, so we just omit the encoding part inside the catch, expecting this exception to never happen as UTF_8 should always be available for URL encoding.

pupitetris committed:
ecp/request: key requests require url-encoding. Also fix for Search

url-encode the key parameter as it is going to be part of the request
URL, which is particularly important for the Lit_ case (for example,
when trying to send a Space).

All url-encoding calls, including that found in SearchRequest changed
to use StandardCharsets.UTF_8.name() or otherwise a MethodNotFound
exception was generated at runtime (maybe only on my development
Android device, but this is more compatible in general). Also,
UnsupportedEncodingException has to be handled, so we just omit the
encoding part inside the catch, expecting this exception to never
happen as UTF_8 should always be available for URL encoding.

wseemannroku commented on src/main/java/com/wseemann/ecp/request/KeyPressRequest.java:
```diff
@@ -10,6 +14,10 @@ final public class KeyPressRequest extends ECPRequest<Void> {
 		super(url);		
+		try {
+		    key = URLEncoder.encode(key, StandardCharsets.UTF_8.name());
+		} catch (UnsupportedEncodingException e) {
```
Can you annotate this constructor as throwing `UnsupportedEncodingException` and let the person implementing the call handle the exception? With this approach the exception is swallowed and it's probably important for the developer to know that we are attempting to encode the key they are passing.

wseemann commented on src/main/java/com/wseemann/ecp/request/KeyPressRequest.java:
```diff
@@ -10,6 +14,10 @@ final public class KeyPressRequest extends ECPRequest<Void> {
 	
 	public KeyPressRequest(String url, String key) {
 		super(url);		
+		try {
```
Can you annotate this constructor as throwing UnsupportedEncodingException and remove the try/catch so the person implementing the call handle the exception? With this approach the exception is swallowed and it's probably important for the developer to know that we are attempting to encode the key they are passing.

pupitetris replied:
OK, I updated the constructors so that they let the exception through.

I also patched RoMote in this PR: https://github.com/wseemann/RoMote/pull/47

pupitetris committed:
request/Key*Request: don't swallow UnsupportedEncodingException

instead, annotate it as a thrown exception, letting the caller decide
what to do with the exception instead of silently swallowing it.
